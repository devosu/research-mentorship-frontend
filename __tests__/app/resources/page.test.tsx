// ./__tests__/app/resources/page.test.tsx
//
// Snapshot tests for the reources page.

// jsdom-jest essential imports.
import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";

// Local testing module imports.
import Resources from "@app/resources/page";

// Behavior for the resources page.
describe("The resources page", () => {
  it("renders a table element", () => {
    const { container } = render(<Resources />);
    expect(container.querySelector("table")).toBeDefined();
  });

  it("matches the snapshot", () => {
    const { container } = render(<Resources />);
    expect(container).toMatchSnapshot();
  });
});
