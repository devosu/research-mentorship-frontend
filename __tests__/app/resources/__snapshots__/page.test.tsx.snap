// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The resources page matches the snapshot 1`] = `
<div>
  <main>
    <header>
      <nav
        class="relative flex w-full flex-wrap items-center justify-between bg-zinc-50 py-2 shadow-dark-mild dark:bg-neutral-700 lg:py-4"
        data-twe-navbar-ref="true"
      >
        <div
          class="flex w-full flex-wrap items-center justify-between px-3"
        >
          <div>
            <a
              class="mx-2 my-1 flex items-center lg:mb-0 lg:mt-0"
              href="/"
            >
              <img
                alt="Logo"
                data-nimg="1"
                decoding="async"
                height="35"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Flogo.png&w=96&q=75"
                srcset="/_next/image?url=%2Fimages%2Flogo.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=96&q=75 2x"
                style="color: transparent;"
                width="35"
              />
            </a>
          </div>
          <button
            aria-controls="navbarSupportedContent4"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="block border-0 bg-transparent px-2 text-black/50 hover:no-underline hover:shadow-none focus:no-underline focus:shadow-none focus:outline-none focus:ring-0 dark:text-neutral-200 lg:hidden"
            data-twe-collapse-init="true"
            data-twe-target="#navbarSupportedContent4"
            type="button"
          >
            <span
              class="[&>svg]:w-7 [&>svg]:stroke-black/50 dark:[&>svg]:stroke-neutral-200"
            >
              <svg
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
          <div
            class="!visible mt-2 hidden flex-grow basis-[100%] items-center lg:mt-0 lg:!flex lg:basis-auto"
            data-twe-collapse-item="true"
            id="navbarSupportedContent4"
          >
            <ul
              class="list-style-none me-auto flex flex-col ps-0 lg:mt-1 lg:flex-row"
              data-twe-navbar-nav-ref="true"
            >
              <li
                class="my-4 ps-2 lg:my-0 lg:pe-1 lg:ps-2"
                data-twe-nav-item-ref="true"
              >
                <a
                  aria-current="page"
                  class="text-black/60 transition duration-200 hover:text-black/80 hover:ease-in-out focus:text-black/80 active:text-black/80 motion-reduce:transition-none dark:text-white/60 dark:hover:text-white/80 dark:focus:text-white/80 dark:active:text-white/80 lg:px-2"
                  data-twe-nav-link-ref="true"
                  href="/"
                >
                  OSU Undergraduate Research Mentorship Program
                </a>
              </li>
            </ul>
            <div
              class="flex items-center"
            >
              <a
                class="me-3 inline-block rounded px-2 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-primary hover:text-primary-600 focus:text-primary-600 focus:outline-none focus:ring-0 active:text-primary-700 dark:text-secondary-600 dark:hover:text-secondary-500 dark:focus:text-secondary-500 dark:active:text-secondary-500"
                data-twe-ripple-color="light"
                data-twe-ripple-init="true"
                href="/"
                type="button"
              >
                Home
              </a>
              <a
                class="me-3 inline-block rounded px-2 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-primary hover:text-primary-600 focus:text-primary-600 focus:outline-none focus:ring-0 active:text-primary-700 dark:text-secondary-600 dark:hover:text-secondary-500 dark:focus:text-secondary-500 dark:active:text-secondary-500"
                data-twe-ripple-color="light"
                data-twe-ripple-init="true"
                href="/resources"
                type="button"
              >
                Resources
              </a>
              <div
                class="relative"
                data-twe-dropdown-ref="true"
              >
                <a
                  aria-expanded="false"
                  class="flex items-center px-2 text-black/60 transition duration-200 hover:text-black/80 hover:ease-in-out focus:text-black/80 active:text-black/80 motion-reduce:transition-none dark:text-white/60 dark:hover:text-white/80 dark:focus:text-white/80 dark:active:text-white/80"
                  data-twe-dropdown-toggle-ref="true"
                  href="#"
                  id="dropdownMenuButton2"
                  type="button"
                >
                  <span
                    class="[&>svg]:h-5 [&>svg]:w-5"
                  >
                    <svg
                      fill="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.75 6a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0zM4.501 20.118a7.5 7.5 0 0114.998 0A17.933 17.933 0 0112 21.75c-2.676 0-5.216-.584-7.499-1.632z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </span>
                  <span
                    class="[&>svg]:w-5"
                  >
                    <svg
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </a>
                <ul
                  aria-labelledby="dropdownMenuButton1"
                  class="absolute z-[1000] float-left m-0 hidden min-w-max list-none overflow-hidden rounded-lg border-none bg-white bg-clip-padding text-left text-base shadow-lg data-[twe-dropdown-show]:block dark:bg-surface-dark"
                  data-twe-dropdown-menu-ref="true"
                >
                  <li>
                    <a
                      class="inline-block rounded px-2 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-primary hover:text-primary-600 focus:text-primary-600 focus:outline-none focus:ring-0 active:text-primary-700 dark:text-secondary-600 dark:hover:text-secondary-500 dark:focus:text-secondary-500 dark:active:text-secondary-500"
                      data-twe-dropdown-item-ref="true"
                      href="/signin"
                    >
                      Sign In
                    </a>
                  </li>
                  <li>
                    <a
                      class="inline-block rounded px-2 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-primary hover:text-primary-600 focus:text-primary-600 focus:outline-none focus:ring-0 active:text-primary-700 dark:text-secondary-600 dark:hover:text-secondary-500 dark:focus:text-secondary-500 dark:active:text-secondary-500"
                      data-twe-dropdown-item-ref="true"
                      href="/signup"
                    >
                      Sign Up
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </header>
    <div
      class="flex flex-col"
    >
      <div
        class="overflow-x-auto sm:-mx-6 lg:-mx-8"
      >
        <div
          class="inline-block min-w-full py-2 sm:px-6 lg:px-8"
        >
          <div
            class="overflow-hidden"
          >
            <table
              class="min-w-full text-left text-sm font-light text-surface dark:text-white"
            >
              <thead
                class="border-b border-neutral-200 font-medium dark:border-white/10"
              >
                <tr>
                  <th
                    class="px-6 py-4"
                    scope="col"
                  >
                    Department Name
                  </th>
                  <th
                    class="px-6 py-4"
                    scope="col"
                  >
                    Department Website
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Biological Chemistry and Pharmacology
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/bcpharm"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Biomedical Education and Anatomy
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/bmea"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Biomedical Informatics
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/biomedical-informatics"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Economics
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://economics.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Cancer Biology and Genetics
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/cancer-biology-and-genetics"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Microbial Infection and Immunity
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/mii"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Neuroscience
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/neuroscience"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Physiology and Cell Biology
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/physiology-and-cell-biology"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Health and Rehabilitation Sciences
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://hrs.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Sports Performance and Injury Prevention
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/sports-medicine/research/research-labs"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Civil, Environmental, and Geodetic Engineering
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://ceg.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Molecular Genetics
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://molgen.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Management and Human Resources
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://fisher.osu.edu/academic-departments/management-human-resources/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Psychology
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://psychology.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Cardiology
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/internal-medicine/cardiovascular-medicine/research"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Psychiatary and Behaviorial Health
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/psychiatry-and-behavioral-health"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Communication
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://comm.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Biomedical Engineering
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://bme.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Pediatrics
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://medicine.osu.edu/departments/pediatrics"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Evolution, Ecology, and Organismal Ecology
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://eeob.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Integrated Systems Engineering
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://ise.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Oncology
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://cancer.osu.edu/for-healthcare-professionals/find-a-department-or-division/department-of-internal-medicine/division-of-medical-oncology"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Veterinary Medicine
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://vet.osu.edu/"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
                <tr
                  class="border-b border-neutral-200 transition duration-300 ease-in-out hover:bg-neutral-100 dark:border-white/10 dark:hover:bg-neutral-600"
                >
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    Operations (Integrated Engineering)
                  </td>
                  <td
                    class="whitespace-nowrap px-6 py-4"
                  >
                    <a
                      href="https://ise.osu.edu/directory/operations-research"
                    >
                      Visit Site
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="bg-zinc-50 text-center dark:bg-neutral-700 lg:text-left"
      style="bottom: 0px; left: 0px; right: 0px; position: fixed;"
    >
      <div
        class="bg-black/5 p-4 text-center text-surface dark:text-white"
      >
        © 2024 Copyright:
        <a
          href="https://usg.osu.edu/"
        >
           
          Undergraduate Student Government at The Ohio State University
        </a>
      </div>
    </footer>
  </main>
</div>
`;
